synapse/rest/admin/devices.py

```diff
@@ -142,6 +142,17 @@ async def on_GET(
             raise NotFoundError("Unknown user")
 
         devices = await self.device_handler.get_devices_by_user(target_user.to_string())
+
+        # mark the dehydrated device by adding a "dehydrated" flag
+        dehydrated_device_info = await self.device_handler.get_dehydrated_device(
+            target_user.to_string()
+        )
+        if dehydrated_device_info:
+            dehydrated_device_id = dehydrated_device_info[0]
+            for device in devices:
+                if device["device_id"] == dehydrated_device_id:
+                    device["dehydrated"] = True
```

Comment:
I think it might be nicer from an Admin API standpoint if `dehydrated` were always present, and was either `true` or `false`?

```suggestion
                is_dehydrated = device["device_id"] == dehydrated_device_id
                device["dehydrated"] = is_dehydrated
```